validator/index.js

```diff
@@ -25,50 +25,233 @@ const path = require('path');
 const program = require('commander');
 const vm = require('vm');
 const url = require('url');
+const util = require('util');
 
 /**
- * If the provided fileOrUrl start with 'http://' or 'https://', downloads
- * with the http or https module, otherwise reads the file with the file module.
- * Or, if the provided fileOrUrl is '-', reads from stdin.
- * Then, calls onSuccess with the data, or onFailure with the error message.
- * This method will check the HTTP status code as well and generate
- * brief error messages if it's not 200.
- * @param {!string} fileOrUrl
- * @param {!function(string)} onSuccess
- * @param {!function(string)} onFailure
+ * Convenience function to detect whether an argument is a ULR or
+ * (perhaps) a local file.
+ * @param {!string} url
+ * @returns {!boolean}
  */
-function readFileOrDownload(fileOrUrl, onSuccess, onFailure) {
-  const digestChunksFrom = function(response) {
+function isHttpOrHttpsUrl(url) {
+  return url.startsWith('http://') || url.startsWith('https://');
+}
+
+/**
+ * Creates a promise which reads from a file.
+ * @param {!string} name
+ * @returns {!Promise<(!string|!Error)>}
+ */
+function readFromFile(name) {
+  return new Promise(function(resolve, reject) {
+    fs.readFile(name, 'utf8', (err, data) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(data);
+      }
+    });
+  });
+}
+
+/**
+ * Creates a promise which reads from a stream.
+ * @param {!stream.Readable} readable
```

Comment:
Here it's lower case:
https://nodejs.org/api/stream.html#stream_class_stream_readable
I wish I had a type checker working for this stuff already, but I couldn't convince the closure compiler yet to do it.